components/dashboard/src/projects/NewProject.tsx

```diff
@@ -42,6 +42,29 @@ export default function NewProject() {
 
     const [authProviders, setAuthProviders] = useState<AuthProviderInfo[]>([]);
 
+    const updateReposInAccounts = useCallback(
```

Comment:
Is it really necessary to use `useCallback` here? I believe we haven't used it elsewhere in the dashboard yet, and if I understand it correctly, calling `updateReposInAccounts` from the `Effect` with the dependency `selectedProviderHost` already ensures that we call it only once every time the dependency changes.

I think I'd prefer not using this feature here, to limit the "magic" in our code. What do you think? 🙂

Reply:
Oh it was actually a cascading effect 🙂

- Without the `useCallback`, we have this linting warning: "The 'updateReposInAccounts' function makes the dependencies of useEffect Hook (at line 203) change on every render. To fix this, wrap the definition of 'updateReposInAccounts' in its own useCallback() Hook"
- Leaving `updateReposInAccounts` out of the dependencies of that `useEffect` to silence the previous warning, we have this other warning: "React Hook useEffect has a missing dependency: 'updateReposInAccounts'. Either include it or remove the dependency array."
- Removing the dependency array altogether to silence the previous warning, of course makes the hook execute on every render, while keeping it empty generates the linting warning about not having empty dependency arrays when we are using some dependencies inside the hook.

The only alternative seems to silence the ESLint rule for this line, but as long as the code still works as intended I'd rather follow the lead of the linting rules and move on.

Let me know what you want me to do!

Reply:
If I'm not mistaken, you've already added the `updateReposInAccounts` closure as a dependency to the `useEffect`. Is that not enough to resolve the warning? (I.e. without needing the extra `useCallback`?)

Reply:
Or, actually I really don't like this pattern of implementing functions that are only called once. What about moving all the `updateReposInAccounts` code back inline into the `useEffect`? (Maybe as an in-line async function if needed.)

Reply:
> What about moving all the updateReposInAccounts code back inline into the useEffect? (Maybe as an in-line async function if needed.)

Oh that's also my usual first move 🤝 but in this case `updateReposInAccounts` is also used elsewhere outside the `useEffect`.
Avoidable? Probably yes, but I think it will come with future refactorings 🔮

Reply:
> If I'm not mistaken, you've already added the `updateReposInAccounts` closure as a dependency to the `useEffect`. Is that not enough to resolve the warning? (I.e. without needing the extra `useCallback`?)

Yes, but we have another warning :D look at the first of my bullet points above.

> Without the useCallback, we have this linting warning: "The 'updateReposInAccounts' function makes the dependencies of useEffect Hook (at line 203) change on every render. To fix this, wrap the definition of 'updateReposInAccounts' in its own useCallback() Hook"

Reply:
That's because without `useCallback` the function get created on every render, and that triggers the dependency array of the `useEffect`